answers.txt

```diff
@@ -1,4 +1,17 @@
 •  What is a gitignore file and what should it include for a Java project? 
+
+The .gitignore file is a text file that specifies which files that Git should ignore while commiting to the repository. 
+In a Java project, we may have classpath files, jar, war, and files that might get added from the IDE we work on etc. Gitignore file can help ignore these files and avoid commiting it to the remote repo
+
 •  What does fast-forward in git mean? 
-•  Why should you have different branches? 
+
+This means that Git will simply refer master to the feature branch's most recent commit if Master hasn't diverged from it.
+
+•  Why should you have different branches?
+
```

Comment:
does having multiple branches cause conflicts when merging?

Reply:
Yes, sometimes it so happens that two developers working on the same file for example push code that overlap, this might create a conflict at the moment, but we can always resolve the conflict by manually going back and validating what is the correct code we need and then continue with the merge. This might be a problem if there are a lot of lines to check before resolving a merge conflict and so the key is to communicate with the team or the developer working on the same file and make separate commits to avoid a merge conflict in the first place.

Reply:
Alright. Understood.